gemini/sample-apps/fashion-trends-prediction/app/data_prep/dataProcessing.py

```diff
@@ -0,0 +1,376 @@
+import base64
+import json
+import os
+import urllib.request
+
+import requests
+import vertexai.preview.generative_models as generative_models
+from config import config
+from genAIprompts import qList, qList2
+from helper_functions_insta import getId
+from vertexai.preview.generative_models import GenerativeModel, Part
+from vertexai.preview.language_models import TextGenerationModel
+from vertexai.preview.vision_models import Image, ImageQnAModel
+
+image_qna_model = ImageQnAModel.from_pretrained("imagetext")
+text_model = TextGenerationModel.from_pretrained("text-bison-32k")
```

Comment:
Is there a reason to use all 3 of these models? Gemini should be able to handle all of these use cases

Reply:
The text_model is used to create article summaries that can be changed to use Gemini. For image captioning, we earlier used the ImageQnA model so we've kept that as a parameter in the code as an alternate to Gemini.

Reply:
Removed the PaLM models. Only Gemini is used now.